### ISSUE TITLE
Health check

### ISSUE THREAD

**Issue by pbroman** [tuub/kitodo-mediaserver#192]
It would be nice to have a health check job, running nightly to check the consistency of the system. This ticket is meant to collect aspects to be checkt by such a job. As for starters:

* All works in the database should have representations in the file system. Since the work path is saved in the database, this should be easy to check. It should only be checked, that the path exists and that the METS xml (named as the work id) is present.
* There should be no unperformed actions in the database older than a configurable amount of days. 

What else do you want? Check, that there are no cached file exceeding a certain date (i.e. that the cache delete job is running)? 

The health check job should in any case be extendable, perhaps simply a list of actions.

**Comment by pbroman** [comment]
Actions are here not the right choice, since they deal with works. We need a new interface for health checks, perhaps simply
`public interface IHealthCheck {
     void check(Map<String, String> parameter);
}`
A health check could then do basically anything it wants.

**Comment by ponchofiesta** [comment]
I agree so far. I think there are a few critical things and some nice to have things. 
- It should be checked if the files from the METS file do exist in the filesystem; the files which are needed for conversion (the whole filegroup).
- (not critically) And vice versa it could be checked if the files in the filesystem are listed in the METS file.
- (not critically) We could check if there are works in the filesystem which do not have an DB entry.
- We could check read and write permissions but this could be difficult from the CLI job. The CLI doesn't know about the correct user that needs to have permissions.
- I think a DB integrity check is a good idea but this needs more considerations.
